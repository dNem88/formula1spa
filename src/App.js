import React, {useEffect, useState} from 'react'
import {HashRouter as Router, Route, Routes} from 'react-router-dom'
import './App.module.css';
import Navigation from './components/common/navigation/Navigation'
import userContext from './context/userContext'
import newsContext from './context/newsContext';
import useAuth from './hooks/useAuth'
import useNews from './hooks/useNews';
import AuthLayout from './components/register/Layout/AuthLayout';
import Register from './components/register/Register/Register';
import Login from './components/register/Login/Login';
import Layout from './components/home/layout/Layout';
import NewsLayout from './components/news/layout/NewsLayout';
import MobileNavigation from './components/common/mobileNavigation/MobileNavigation';
import Footer from './components/common/footer/Footer'
import ActiveNews from './components/news/activeNews/ActiveNews';
import VideoLayout from './components/video/videoLayout/VideoLayout'
import MainVideo from './components/video/mainVideo/MainVideo'
import StandingsLayout from './components/standings/standingsLayout/StandingsLayout';
import ScheduleLayout from './components/schedule/ScheduleLayout/ScheduleLayout';
import DriversStandings from './components/home/standings/DriversStandings';
import ConstructorsStandings
 from './components/common/constructorsStandings/ConstructorsStandings';
function App() {
  const [viewport, setViewport] = useState({width: window.innerWidth})
  let contextNews = useNews();
  let context = useAuth()
  
  function onResize(e) {
    setViewport({width: window.innerWidth})
  }
  useEffect(() => {
    if (context.isLogged === 'initial') {
       context.verify()
    }
  }, [])
  useEffect(() => {
     window.addEventListener('resize', onResize)
     return () => {
        window.removeEventListener('resize', onResize)
     }
  }, [])
  console.log('RENDER FROM APP>JS')
  return (
    <Router>
      <userContext.Provider value={context}>
        {viewport.width > 850 ? <Navigation/> : <MobileNavigation/>}
      </userContext.Provider>
      <main>
        <Routes>
          <Route path={'/'} element={
            <newsContext.Provider value={contextNews}>
              <Layout/>
            </newsContext.Provider>}/>
          <Route path={'/news/*'} element={
            <newsContext.Provider value={contextNews}>
              <NewsLayout/>
            </newsContext.Provider>}>
              <Route path={':id'} element={<ActiveNews index={false}/>}/>
              <Route index element={
                <ActiveNews index={true}/>}/>
          </Route>
          <Route path={'/videos/*'} element={<VideoLayout/>}>
                <Route path=":id" element={<MainVideo/>}/>
                <Route index element={<MainVideo/>}/>
          </Route>
          <Route path={'/schedule'} element={<ScheduleLayout/>}/>
          <Route path={'/standings/*'} element={<StandingsLayout/>}>
                <Route path={'drivers'} element={<DriversStandings count={25}/>}/>
                <Route path={'constructors'} element={<ConstructorsStandings/>}/>
                 <Route index element={<DriversStandings count={25}/>}/>
          </Route>
          <Route path={'/drivers'} element={<p>Drivers Page</p>}/>
          <Route path={'/teams'} element={<p>Teams Page</p>}/>
          <Route path={'/auth'} element={<AuthLayout/>}>
            <Route path={'login'} element={
              <userContext.Provider value={context}>
                <Login/>
              </userContext.Provider>
              }/>
            <Route path={'register'} element={
              <userContext.Provider value={context}>
                <Register/>
              </userContext.Provider>
            }/>
            <Route index element={<userContext.Provider value={context}>
                <Login/>
              </userContext.Provider>}/>
          </Route>
          <Route path={'/profile'} element={<p>Profile Page</p>}/>
        </Routes>
      </main>
      <Footer/>
    </Router>
  );
}

export default App;
