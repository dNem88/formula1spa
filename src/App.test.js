import renderer from 'react-test-renderer';
import App from './App';
import React from 'react'

describe('test App component', () => {
  it('Should render app compoent', () => {
    let app = renderer.create(<App/>).toJSON()
    expect(app).toMatchSnapshot()
    expect(app[0].type).toBe('nav')
    expect(app[1].type).toBe('main')
    expect(app[2].type).toBe('footer')
  })
})