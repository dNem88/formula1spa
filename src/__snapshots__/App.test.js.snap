// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test App component Should render app compoent 1`] = `
Array [
  <nav
    className="nav"
  >
    <div
      className="logo-container"
    >
      <img
        src="authLogo.svg"
      />
    </div>
    <div
      className="links-container"
    >
      <a
        href="#/"
        onClick={[Function]}
      >
        Home
      </a>
      <a
        href="#/news"
        onClick={[Function]}
      >
        News
      </a>
      <a
        href="#/videos"
        onClick={[Function]}
      >
        Videos
      </a>
      <a
        href="#/schedule"
        onClick={[Function]}
      >
        Schedule
      </a>
      <a
        href="#/standings"
        onClick={[Function]}
      >
        Standings
      </a>
      <a
        href="#/drivers"
        onClick={[Function]}
      >
        Drivers
      </a>
      <a
        href="#/teams"
        onClick={[Function]}
      >
        Teams
      </a>
    </div>
    <div
      className="auth-container"
    />
  </nav>,
  <main>
    <section
      className="trending"
    >
      <div
        className="lds-ring"
      >
        <div />
        <div />
        <div />
        <div />
      </div>
    </section>
    <section
      className="news"
    >
      <div
        className="lds-ring"
      >
        <div />
        <div />
        <div />
        <div />
      </div>
    </section>
    <fieldset
      className="editor-pick"
    >
      <legend
        className="editor-legend"
      >
        EDITOR'S PICK
      </legend>
      <div
        className="lds-ring"
      >
        <div />
        <div />
        <div />
        <div />
      </div>
    </fieldset>
    <h2
      className="h-news"
    >
      More news
    </h2>
    <section
      className="more-news"
    >
      <div
        className="lds-ring"
      >
        <div />
        <div />
        <div />
        <div />
      </div>
    </section>
    <div
      className="lds-ring"
    >
      <div />
      <div />
      <div />
      <div />
    </div>
    <section
      className="standings"
    >
      <div
        className="main-container"
      >
        <div
          className="heading-container"
        >
          <h2>
            DRIVERS STANDINGS
          </h2>
        </div>
        <div
          className="standings-container"
        >
          <div
            className="lds-ring"
          >
            <div />
            <div />
            <div />
            <div />
          </div>
        </div>
        <a
          href="#/standings"
          onClick={[Function]}
        >
          VIEW FULL STANDINGS
        </a>
      </div>
    </section>
  </main>,
  <footer
    className="footer"
  >
    <div
      className="footer-container"
    >
      <a
        href="#/"
        onClick={[Function]}
      >
        Home
      </a>
      <a
        href="#/news"
        onClick={[Function]}
      >
        News
      </a>
      <a
        href="#/videos"
        onClick={[Function]}
      >
        Videos
      </a>
      <a
        href="#/schedule"
        onClick={[Function]}
      >
        Schedule
      </a>
      <a
        href="#/standings"
        onClick={[Function]}
      >
        Standings
      </a>
      <a
        href="#/drivers"
        onClick={[Function]}
      >
        Drivers
      </a>
      <a
        href="#/teams"
        onClick={[Function]}
      >
        Teams
      </a>
    </div>
    <div
      className="strong-container"
    >
      <strong>
        @Formula 1 React Project 2022
      </strong>
    </div>
  </footer>,
]
`;
